### PR TITLE
A pure numpy version for yolov5_trt.py

### PR DESCRIPTION
**Pull request by HaiyangPeng**
- For decoupling the dependency of PyTorch, this code provides a pure numpy manner to write nms operation in yolov5_trt.py. I think it can offer simple and convenient TensorRT experience for yolov5.
- In detail, the following functions and codes were modified with the corresponding reason:
1.`pred = torch.Tensor(pred).cuda()` was deleted for saving GPU memory
2.`xywh2xyxy()` deleted the option of Tensor
3.`torchvision.ops.nms(boxes, scores, iou_threshold=IOU_THRESHOLD).cpu()` is replaced by a pure numpy code `self.non_max_suppression(pred, origin_h, origin_w, conf_thres=CONF_THRESH, nms_thres=IOU_THRESHOLD)`. This code contains the process of filtering detections with low confidence and common nms. 
4.`bbox_iou()` is added to compute the iou of two boxes for filtering the detection with high iou. The detections are sorted by confs with descending order.
5.The overall code structure is similar to yolov5_trt.py and the other codes there have not been modified. I have tested this code on my own dataset successfully, and if there are any bugs, contact me. I'm happy to fix them, THX.
In addition, the nms in yolov5_trt.py has no weighting ops, such as iou and confidence, which is not in accordance with git@github.com:ultralytics/yolov5.git. To get better detections, these ops can be included. Since the tensorRT engine output is (x, y, w, h, conf, cls_id), not (x, y, w, h, conf, prob1, prob2, ...), it is unavailable to implement these ops at this time.
Other question:
Is it a need to change the padding color of (128, 128, 128) to (114, 114, 114) in the `preprocess_image` function, since the training code in git@github.com:ultralytics/yolov5.git uses (114, 114, 114). BTW, for my case, this does not affect the result ^ - ^

### CODE REVIEW

**Comment by wang-xinyu** [comment]
Nice, this is helpful.

Could you re-submit the PR? Seems you have included hrnet_trt.py in this PR. which is already be merged into master.
